stdlib/private/StdlibCollectionUnittest/CheckMutableCollectionType.swift.gyb

```diff
@@ -907,16 +907,20 @@ func checkPartition_${'Predicate' if predicate else 'WhereElementIsComparable'}(
 
   var c = makeWrappedCollectionWithComparableElement(elements)
 %   end
-
+  let pivotElt  = c.first
+  var pivot = c.startIndex
 %   if predicate:
   let closureLifetimeTracker = LifetimeTracked(0)
-  let pivot = c.partition() {
-    (lhs, rhs) in
-    _blackHole(closureLifetimeTracker)
-    return extract(lhs).value < extract(rhs).value
+  if let first = pivotElt {
+    pivot = c.partition(by: {
```

Comment:
Why not try calling `partition` on empty collections?  That's important to test.  Feel free to unconditionally unwrap `pivotElt` in the closure, it should never get executed.